content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "From code search toward a code intelligence platform"
+publishDate: 2022-09-27T10:00-07:00
+description: Finding things in code is good, but learning things about code is great. Sourcegraph's code intelligence platform is built to help.
+tags: [blog]
+slug: "code-search-to-code-intelligence"
+published: false
+heroImage: 
+socialImage: 
+---
+
+No matter how good your other dev tools, coding is still complex. Even if you've configured your editor just right, sped up your test suite, set up continuous deployment, added observability, etc., coding is still complex. Higher-level languages, libraries, frameworks, cloud providers, etc., that encapsulate complexity just let you spend your complexity budget elsewhere. The overall complexity of code has increased, not declined, over time, and there's no sign of that trend changing.
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc. 
+
+On top of this, devs are more important to their companies than ever. Devs have gone from being viewed as cost centers to being seen as the center of product development, so the issue of complexity is a business problem just as much as a dev problem.
```

Comment:
```suggestion
On top of this, devs are more important than ever. Devs have gone from being viewed as cost centers to being seen as the center of product development, so the issue of complexity is a business problem just as much as a dev problem.
```

Reply:
Fine to cut the whole phrase or just the word "their" but it seems odd tonally to me to say to an audience of presumably devs and dev leaders "you're all important to your companies" – seems more like consulting-y, non-dev speak to me to say that. Imagining myself addressing or being in the audience of devs in person and someone says "you devs are important to your companies" sounds too company-focused IMO.